Modules/mathmodule.c

```diff
@@ -3914,7 +3916,65 @@ math_nextafter_impl(PyObject *module, double x, double y)
         return PyFloat_FromDouble(y);
     }
 #endif
-    return PyFloat_FromDouble(nextafter(x, y));
+    // fast path:
+    if (steps == 1) {
+        return PyFloat_FromDouble(nextafter(x, y));
+    }
+    if (steps < 0) {
+        PyErr_SetString(PyExc_ValueError, "steps must be >= 0");
+        return NULL;
+    }
+    if (steps == 0)
+        return PyFloat_FromDouble(x);
+    if (Py_IS_NAN(x) || Py_IS_NAN(y))
+        return PyFloat_FromDouble(x+y);
+
+    uint64_t usteps = steps;
+
+    union pun {double f; uint64_t i;};
```

Comment:
Please add a comment noting that this code assumes that `double` and `uint64_t` have matching endianness. (There's nothing in the C standards that guarantees this, though it's been a while since I've seen a platform that violates it. The most likely form of violation would be a "mixed-endian" double.)